### PR TITLE
jsk_perception: add face_pose_estimation

### PR DESCRIPTION
**Pull request by furushchev**
- Add detector for face pose/gender of people

### CODE REVIEW

**Comment by furushchev** [comment]
waiting #2205

File: jsk_perception/node_scripts/face_pose_estimation.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+        orig_h, orig_w, _ = img.shape
+        assert orig_h > 0 and orig_w > 0
+
+        # transform image
+        img = img.astype(np.float32) / 255.0
+        img = cv2.resize(img, (227, 227))
+        img = cv2.normalize(img, None, -0.5, 0.5, cv2.NORM_MINMAX)
+        img = np.transpose(img, (2, 0, 1))  # CHW
+        img = img[None]  # single batch
```
Comment (furushchev) [inline jsk_perception/node_scripts/face_pose_estimation.py]: Memo for myself: does this mean this network supports multi images as input?

File: jsk_perception/launch/face_pose_estimation.launch [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <remap from="~input" to="$(arg INPUT_IMAGE)"/>
+    <remap from="~input/pose_2d" to="people_pose_estimation_2d/pose_2d" />
+    <remap from="~input/pose" to="people_pose_estimation_2d/pose" />
+    <rosparam subst_value="true">
+      gpu: $(arg GPU)
+    </rosparam>
+  </node>
+
+  <node name="draw_rects"
+        pkg="jsk_perception" type="draw_rect_array.py">
```
Comment (furushchev) [inline jsk_perception/launch/face_pose_estimation.launch]: This should be updated.

**Comment by k-okada** [comment]
https://travis-ci.org/jsk-ros-pkg/jsk_recognition/jobs/293018430

```
-- run_tests.py: execute commands
  /opt/ros/indigo/share/rostest/cmake/../../../bin/rostest --pkgdir=/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception --package=jsk_perception --results-filename test_color_histogram.xml --results-base-dir /workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/color_histogram.test 
... logging to /workspace/jsk-ros-pkg/jsk_recognition/log/rostest-763abbfd5394-37535.log
[ROSUNIT] Outputting test results to /workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/rostest-test_color_histogram.xml
Traceback (most recent call last):
  File "/opt/ros/indigo/share/image_view2/test/publish_mouse_event.py", line 6, in <module>
    from scipy.misc import lena
ImportError: cannot import name lena
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
Traceback (most recent call last):
  File "/opt/ros/indigo/share/image_view2/test/publish_mouse_event.py", line 6, in <module>
    from scipy.misc import lena
ImportError: cannot import name lena
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
Traceback (most recent call last):
  File "/opt/ros/indigo/share/image_view2/test/publish_mouse_event.py", line 6, in <module>
    from scipy.misc import lena
ImportError: cannot import name lena
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
Traceback (most recent call last):
  File "/opt/ros/indigo/share/image_view2/test/publish_mouse_event.py", line 6, in <module>
    from scipy.misc import lena
ImportError: cannot import name lena
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
testtest_color_histogram ... ok
[ROSTEST]-----------------------------------------------------------------------
[jsk_perception.rosunit-test_color_histogram/test_published][FAILURE]-----------
Topic [/color_histogram/intensity_histogram] is not published
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/opt/ros/indigo/share/jsk_tools/src/test_topic_published.py", line 96, in test_published
    checker.topic_name)
  File "/usr/lib/python2.7/unittest/case.py", line 424, in assertTrue
    raise self.failureException(msg)
--------------------------------------------------------------------------------
SUMMARY
 * RESULT: FAIL
 * TESTS: 1
 * ERRORS: 0
 * FAILURES: 1
```

**Comment by furushchev** [comment]
@k-okada Thanks for report. Fixed. But I'm not sure why it fails from **now**...

**Pull request merged**